### PR TITLE
Plans step redesign: Implement the header section of each plan

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR implements the experiment that redesigns the signup flow plans step. Users in the treatment group will see the new header section for each plan in the signup plans step.

NOTE: 
1. This PR uses the changes in https://github.com/Automattic/wp-calypso/pull/49877 as the base branch and is built on top of it.
2. This PR only implements the header section. The rest of the plan sections are implemented in subsequent PRs.

**Important links**
1. [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813)
2. pau2Xa-2yE-p2 for the project thread.

**Annual plans header:**

Screenshot 2021-02-19 at 11 01 54 AM

**Monthly plans header:**

Screenshot 2021-02-19 at 11 02 02 AM

#### Testing instructions

* Open http://calypso.localhost:3000/ and assign yourself to treatment of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Begin signup at `/start`and verify that the plans step UI matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813). Verify for both free subdomain and custom domain selections.
* Verify that the UI matches the Figma design for both annual and monthly plan views.
* Confirm that in Calypso's /plans view, the UI is unchanged. 
* Also confirm that the launch flow UI is unchanged.
* Verify that `control` experience is unchanged. 

Related to https://github.com/Automattic/wp-calypso/pull/49877

### CODE REVIEW

File: client/my-sites/plan-features-comparison/header.jsx [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+		if ( isInSignup && isMonthlyPlan && annualPricePerMonth < rawPrice ) {
+			const discountRate = Math.round( ( 100 * ( rawPrice - annualPricePerMonth ) ) / rawPrice );
+			return translate( `Save %(discountRate)s%% by paying annually`, { args: { discountRate } } );
+		}
+
+		if ( isInSignup && ! isMonthlyPlan ) {
+			const annualPriceObj = getCurrencyObject( rawPriceAnnual, currencyCode );
+			const annualPriceText = `${ annualPriceObj.symbol }${ annualPriceObj.integer }`;
+
+			return translate( 'billed as %(price)s annually', {
```
Comment (a8ci18n) [inline client/my-sites/plan-features-comparison/header.jsx:124]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'billed as %s annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/feature-comparison-signup-plans)

**Review by dzver — COMMENTED** [review]
I followed the test plan and I observed the following:

1. The launch-site flow in currencies where first-year promotional pricing is present look a bit unexpected:

Screenshot 2021-02-16 at 19 38 33

vs this in Trunk:

Screenshot 2021-02-16 at 20 01 08


2. The mobile view is missing in treatment

Screenshot 2021-02-16 at 19 32 49

There's some logic for discounts, is it exactly the same as the existing logic?

**Comment by niranjan-uma-shankar** [comment]
> The launch-site flow in currencies where first-year promotional pricing is present look a bit unexpected:

Nice catch, I've fixed this.

Also, I've removed the promotional pricing striked out UI for treatment in these currencies, because the "You're saving ...." copy shown for annual prices doesn't fit well when the promotional pricing is shown. Looks like we hide the striked out UI in new onboarding flow too, so keeping the same behaviour here.

Screenshot 2021-02-19 at 11 01 54 AM

**Review by dzver — APPROVED** [review]
OK, but in case we launch in INR/PHP/MXN this will have to be revisited.

The issue is that there's no mention that the renewal price will be different. The checkout also seems to suffer from this problem so the actual price will not be visible anywhere.

Screenshot 2021-02-19 at 18 19 45 The actual cost is 4200. I don't see it anywhere.

So I think this design is only fine because we'll launch the experiment for en/us, which will see USD pricing.

**Pull request merged**